### ISSUE TITLE
cabal complains about truncated .tar archives

### ISSUE THREAD

**Issue by edsko** [well-typed/hackage-security#112]
```
$ cabal fetch bytestring-plain
Resolving dependencies...
Downloading bytestring-plain-0.1.0.2...
truncated tar archive
```

(this yields an empty hackage.haskell.org/bytestring-plain/0.1.0.2/ directory)

another example

```
$ cabal-secure-beta install apiary
Resolving dependencies...
Downloading apiary-1.4.5...
truncated tar archive
```

I wonder if this is related to the bug in the tar index building which is also affecting the mirror client.

**Comment by edsko** [comment]
Ok, `cabal-secure-beta Ebnf2ps-1.0.15` also results in this error, so I suspect this may be due to the tar index problem (#109).

**Comment by edsko** [comment]
Hmmm, perhaps this is a different problem after all.. I'll need some more digging.

**Comment by edsko** [comment]
Sorry, my testing was getting confused by an orthogonal bug (#114). When I don't use the tar index in `getFromIndex` (and just search the tarball) this "truncated index" thing disappears. Rebuilding the tar index just before looking up the file doesn't help, just yields the same error.

**Comment by edsko** [comment]
Reproduced this using the `tar` package only.

**Comment by edsko** [comment]
This was due to a bug in the tar package. Fixed, but still needs to be released (#121).
